### ISSUE TITLE
Empty lines around method definition.

### ISSUE THREAD

**Issue by legendetm** [bbatsov/rubocop#3452]
I would like to configure rubocop to check the following code style 

``` ruby
# file1.rb
module Providers
  class ProviderA
    class Xml
      include Provides::Xml

      def to_xml
        ...
      end

    end
  end
end

# file2.rb
class XmlBuilder

  def tag!(name, options)
  end

end
```

Basically this **cop** requires empty lines around `def .. end` method definitions and should be checked in classes and modules.
## With default configuration

``` yml
Style/EmptyLinesAroundClassBody:
  EnforcedStyle: no_empty_lines
Style/EmptyLinesAroundModuleBody:
  EnforcedStyle: no_empty_lines
```

the output is

```
file1.rb:8:1: C: Extra empty line detected at class body end.
file2.rb:2:1: C: Extra empty line detected at class body beginning.
file2.rb:5:1: C: Extra empty line detected at class body end.
```

and the valid code style is

``` ruby
# file1.rb
module Providers
  class ProviderA
    class Xml
      include Provides::Xml

      def to_xml
      end
    end
  end
end

# or file1.rb
module Providers
  class ProviderA
    class Xml
      include Provides::Xml
      def to_xml
      end
    end
  end
end

# file2.rb
class XmlBuilder
  def tag!(name, options)
  end
end
```
## **And with configuration**

``` yml
Style/EmptyLinesAroundClassBody:
  EnforcedStyle: empty_lines
Style/EmptyLinesAroundModuleBody:
  EnforcedStyle: empty_lines
```

the output is

```
file1.rb:2:1: C: Empty line missing at module body beginning.
  class ProviderA
^
file1.rb:3:1: C: Empty line missing at class body beginning.
    class Xml
^
file1.rb:4:1: C: Empty line missing at class body beginning.
      include Provides::Xml
^
file1.rb:10:1: C: Empty line missing at class body end.
  end
^
file1.rb:11:1: C: Empty line missing at module body end.
end
^
```

and the valid code style is

``` ruby
# file1.rb
module Providers

  class ProviderA

    class Xml

      include Provides::Xml

      def to_xml
      end

    end

  end

end

# file2.rb
class XmlBuilder

  def tag!(name, options)
  end

end
```

This configuration requires too many empty lines in case of nested classes and modules.

**Comment by bbatsov** [comment]
Well, I think this style was proposed a while back, but no one was interested in implementing it. I'm open to supporting it, but someone else will have to implement it.
